chain/src/types.rs

```diff
@@ -193,24 +193,26 @@ pub struct TxHashSetRoots {
 }
 
 impl TxHashSetRoots {
-	/// Accessor for the underlying output PMMR root
-	pub fn output_root(&self) -> Hash {
-		self.output_roots.output_root()
+	/// Accessor for the output PMMR root (rules here are block height dependent).
+	/// We assume the header version is consistent with the block height and that this
+	/// has been validated elsewhere.
```

Comment:
Better to specifically state that it's validate_header()'s responsibility to check for correct header version.